### PR TITLE
sushi fileshare

### PR DESCRIPTION
**Pull request by jackhaibo**
**What this PR does / why we need it**:
File share implementation of sushi.
The main change is that I decoupled sushi into file share and block.

### CODE REVIEW

File: csi/server/plugin/opensds/constants.go [hunk 1]
```diff
@@ -15,66 +15,86 @@
 package opensds
 
 // K8s storage class parameter keywords
-
 const (
-	KParamProfile           = "profile"
-	KParamAZ                = "availabilityzone"
-	KParamEnableReplication = "enablereplication"
-	KParamSecondaryAZ       = "secondaryavailabilityzone"
+	ParamProfile           = "profile"
```
Comment (leonwanghui) [inline csi/server/plugin/opensds/constants.go]: Why do you update all constants here?

File: csi/server/plugin/opensds/constants.go [hunk 2]
```diff
@@ -15,66 +15,86 @@
...
+	// default filesystem type
+	DefFSType               = "ext4"
+	DefaultAvailabilityZone = "default"
 )
 
 // Csi configuration parameters and values
 const (
 	// parameters
-	KCSIVolumeMode = "CSIVolumeMode"
+	CSIVolumeMode = "CSIVolumeMode"
```
Comment (leonwanghui) [inline csi/server/plugin/opensds/constants.go]: Same here

File: csi/server/plugin/opensds/constants.go [hunk 3]
```diff
@@ -15,66 +15,86 @@
...
-	KVolumeReplicationId = "replicationId"
-	kStorageType         = "storagetype"
+	VolumeName          = "name"
+	VolumeStatus        = "status"
+	VolumeAZ            = "availabilityZone"
+	VolumePoolId        = "poolId"
+	VolumeProfileId     = "profileId"
+	VolumeLvPath        = "lvPath"
+	VolumeReplicationId = "replicationId"
+	StorageType         = "storageType"
```
Comment (leonwanghui) [inline csi/server/plugin/opensds/constants.go]: why rename to `storageType`, I remembered you convert it to all lower cases.

File: csi/server/main.go [hunk 4]
```diff
@@ -48,6 +49,9 @@ func main() {
 	// opensds auth strategy
 	flag.StringVar(&opensdsAuthStrategy, "opensdsAuthStrategy", util.OpensdsDefaultAuthStrategy, "OpenSDS Auth Strategy")
 
+	// storageType file or block
+	flag.StringVar(&opensdsStorageType, "storageType", util.OpensdsDefaultAuthStrategy, "OpenSDS Auth Strategy")
```
Comment (leonwanghui) [inline csi/server/main.go]: Why the default value here is util.OpensdsDefaultAuthStrategy

File: csi/server/plugin/opensds/plugin.go [hunk 5]
```diff
@@ -16,32 +16,33 @@ package opensds
...
 type Plugin struct {
-	Client *client.Client
+	PluginStorageType string
+	Client            *client.Client
+	VolumeClient      *Volume
+	FileShareClient   *FileShare
 }
 
-func NewServer(endpoint, authStrategy string) (plugin.Service, error) {
+func NewServer(endpoint, authStrategy, storageType string) (plugin.Service, error) {
```
Comment (leonwanghui) [inline csi/server/plugin/opensds/plugin.go]: I don't know why storageType is required to start a new server, please elaborate.

File: csi/server/plugin/opensds/controller.go [hunk 6]
```diff
@@ -594,50 +184,11 @@ func (p *Plugin) ControllerUnpublishVolume(
...
-	for _, act := range acts {
-		if ok := UnpublishAttachmentList.isExist(act.Id); !ok {
-			glog.Infof("add attachment id %s into unpublish attachment list", act.Id)
-			UnpublishAttachmentList.Add(act)
-			UnpublishAttachmentList.PrintList()
-		}
-	}
-
-	return &csi.ControllerUnpublishVolumeResponse{}, nil
+	return p.FileShareClient.ControllerUnPublishFileShare(req)
```
Comment (leonwanghui) [inline csi/server/plugin/opensds/controller.go]: s/ControllerUnPublishFileShare/ControllerUnpublishFileShare

Comment (jackhaibo) [reply csi/server/plugin/opensds/controller.go]: Done.

File: csi/server/plugin/opensds/plugin.go (hunk 5 above)
Comment (jackhaibo) [reply csi/server/plugin/opensds/plugin.go]: Done.

**Comment by leonwanghui** [comment]
@jackhaibo Could you have a test for this feature?

**Pull request merged**